tests/test_process.py

```diff
@@ -1,67 +1,242 @@
-# import pytest
-# import agoradatatools.process as process
-# import pandas as pd
-
-# file_object = {
-#     "id": "syn25838546",
-#     "format": "table",
-#     "final_filename": "teams.json",
-#     "provenance": ['syn25838546'],
-#     "destination": "syn25871921"
-# }
-
-# dataset_object_multi = {
-#         "team_info": {
-#             "files": [
-#                 {
-#                     "syn12615624": "csv"
-#                 },
-#                 {
-#                     "syn12615633": "csv"
-#                 }
-#             ],
-#             "final_format": "json",
-#             "additional_transformations": [
-#                 {
-#                     "join_datasets": {
-#                         "left": "syn12615624",
-#                         "right": "syn12615633",
-#                         "how": "left",
-#                         "on": "team"
-#                     }
-#                 }
-#             ],
-#             "provenance": [
-#                 "syn12615624",
-#                 "syn12615633"
-#             ],
-#             "destination": "syn25871921"
-#         }
-#     }
-
-# dataset_object_single = {
-#         "neuropath_corr": {
-#             "files": [
-#                 {
-#                     "syn22017882": "csv"
-#                 }
-#             ],
-#             "final_format": "json",
-#             "provenance": [
-#                 "syn22017882"
-#             ],
-#             "column_rename": {
-#                 "ensg": "ensembl_gene_id",
-#                 "gname": "hgnc_gene_id"
-#             },
-#             "destination": "syn25871921"
-#         }
-#     }
-
-# def test_process_dataset():
-#     good_result = process.process_dataset(dataset_obj=dataset_object_single)
-#     assert type(good_result) is tuple
-
-# def test_create_data_manifest():
-#     assert type(process.create_data_manifest(parent='syn27406021')) == pd.DataFrame
-#     assert process.create_data_manifest() is None
+import pandas as pd
+from unittest.mock import patch
+import argparse
+
+from agoradatatools.etl import extract, transform, load, utils
+from agoradatatools import process
+
+
+dataset_object = {
+    "neuropath_corr": {
+        "files": [{"name": "test_file_1", "id": "syn1111111", "format": "csv"}],
+        "final_format": "json",
+        "provenance": ["syn1111111"],
+        "destination": "syn1111113",
+    }
+}
+
+dataset_object_col_rename = {
+    "neuropath_corr": {
+        "files": [{"name": "test_file_1", "id": "syn1111111", "format": "csv"}],
+        "final_format": "json",
+        "provenance": ["syn1111111"],
+        "destination": "syn1111113",
+        "column_rename": {"col_1": "new_col_1", "col_2": "new_col_2"},
+    }
+}
+
+
+def test_process_dataset_with_column_rename():
```

Comment:
As I mentioned in the PR description, I was looking for a way to create reusable mocking for the dependent functions in each of the larger functions I am testing. Especially for `process_dataset()` and `process_all_files()`. From what I understand, the two best ways are using fixtures or classes. I thought that a model where each function in `process.py` has a class for it with the reusable mocked functions being created using the `setUp` method of [unittest](https://docs.python.org/3/library/unittest.html), but I was unable to get it working. Just looking for feedback on which direction might be best to spend more time trying to implement (if either is even necessary for this case).

Reply:
There is also a package that Bruno uses called `mocker` that [simplifies the context manager.](https://github.com/Sage-Bionetworks-Workflows/sagetasks/blob/ea06f56944f7a6531d022738f47511dd5727c448/tests/nextflowtower/test_utils.py#L65-L66)

That said, I personally prefer having less dependencies, but just using mocker alone doesn't resolve the fact that this function requires the patching of many other functions.

Reply:
I did try something similar to your class example, but I wasn't able to get the patched objects defined in `setup_method` to pass to the other functions of the class.

Reply:
@BWMac Ah, my apologies, I forgot to add the following:

```
    def setup_method(self):
        self.patch_get = patch.object(extract, "get_entity_as_df").start()
        self.patch_standardize_col = patch.object(transform, "standardize_column_names").start()
        self.patch_standardize_val = patch.object(transform, "standardize_values").start()
        self.patch_rename = patch.object(transform, "rename_columns").start()
        self.patch_df_to_json = patch.object(load, "df_to_json").start()
        self.patch_load = patch.object(load, "load").start()
    
    def teardown_method(self):
        self.patch_get.stop()
        self.patch_standardize_col.stop()
        self.patch_standardize_val.stop()
        self.patch_rename.stop()
        self.patch_df_to_json.stop()
        self.patch_load.stop()
```

Reply:
Oh, that makes sense. Yeah that seems to be what was missing. Thanks